### PR TITLE
EP-4143: Allow user to override default description in affiliate product cards

### PR DESCRIPTION
**Pull request by Mwindo**
Also adding conditions for displaying Banner Text and Description Text controls.

https://user-images.githubusercontent.com/47676832/206857182-1b8963e3-8fe0-4ded-b945-a0b636d6af54.mov

### CODE REVIEW

File: affiliate/blocks/productCard/src/ProductCard.jsx [hunk 1]
```diff
@@ -1,20 +1,31 @@
...
-  return Object.entries(options).map(([key, value]) => `${key}=${value}`).join(',');
+function serializeOptions(
+    displayDescription,
+    displayImage,
+    description,
+) {
+  const filteredOptions = { displayDescription, displayImage };
+  if (displayDescription) {
+    filteredOptions.description = description;
+  }
```
Comment (Mwindo) [inline affiliate/blocks/productCard/src/ProductCard.jsx:12]: Cleaner to not pass this if we aren't displaying the description.

File: affiliate/blocks/productCard/src/ProductCard.jsx [hunk 2]
```diff
@@ -1,20 +1,31 @@
...
+  return Object.entries(filteredOptions)
+      .filter(([, value]) => (value !== '' && value != null))
```
Comment (Mwindo) [inline affiliate/blocks/productCard/src/ProductCard.jsx:14]: Also cleaner not to pass any parameter that is blank.

File: affiliate/blocks/productCard/src/ProductCard.jsx [hunk 3]
```diff
@@ -1,20 +1,31 @@
...
+  displayDescription,
+  displayImage,
+  description,
+) {
+  const filteredOptions = { displayDescription, displayImage };
+  if (displayDescription) {
+    filteredOptions.description = description;
+  }
+  return Object.entries(filteredOptions)
+    .filter(([, value]) => (value !== '' && value != null))
```
Comment (Mwindo) [inline affiliate/blocks/productCard/src/ProductCard.jsx:14]: Also cleaner not to pass any blank parameters.

**Review by amittman — APPROVED** [review]
LGTM

**Comment by Mwindo** [comment]
Thanks for the quick review @amittman! I'll merge but wait to release until Monday.

**Pull request merged**